shareable url

ascott opened the pull request:
I started using this chrome plugin the other day to hangout with my sister in Vancouver. A shareable url would be really handy.

Fixes issue https://github.com/boyers/netflixparty-chrome/issues/1

![netflix-party-sharing](https://cloud.githubusercontent.com/assets/130878/12934554/445eabca-cf44-11e5-9b7d-7bcb78a8addc.gif)

boyers commented on popup.html:
```diff
@@ -27,6 +27,8 @@
         <h3>Session ID:</h3>
         <input id="session-id-info" type="text" readonly="true" autocomplete="off" autofocus />
+        <h3>Share Url:</h3>
+        <p id="share-url"></p>
```
Two suggestions:

1. `Share Url` -> `Share URL`
2. Use this for the URL:

```html
<input id="share-url" type="text" readonly="true" />
```

boyers commented on popup.js:
```diff
@@ -78,6 +79,17 @@ $(function() {
 
+        // if there is a session id in the url, join that session.
+        var sessionIdFromUrl = getSessionIdFromUrl(tabs[0].url);
+        if(sessionIdFromUrl) {
```
nit, space after `if`

boyers commented on popup.js:
```diff
@@ -133,3 +145,19 @@ $(function() {
 });
+
+function showShareUrl(sessionId) {
+  var baseUrl;
```
unused?

boyers commented on popup.js:
```diff
@@ -133,3 +145,19 @@ $(function() {
+  var baseUrl;
+  chrome.windows.getCurrent(function(w) {
+    chrome.tabs.getSelected(w.id, function (response){
+      var url = response.url.split('?')[0]
```
nit, terminate with semicolon

boyers commented on popup.js:
```diff
@@ -133,3 +145,19 @@ $(function() {
+function showShareUrl(sessionId) {
+  var baseUrl;
+  chrome.windows.getCurrent(function(w) {
+    chrome.tabs.getSelected(w.id, function (response){
```
nit, space before `{` and remove space after `function` (for consistency with existing code)

boyers commented on popup.js:
```diff
@@ -133,3 +145,19 @@ $(function() {
+function getSessionIdFromUrl(url) {
+  if(url.includes('?npSessionId=')) {
+    return url.split('?npSessionId=')[1];
+  }
```
Hmm, can we make this code more robust by allowing other query params in the URL? I know the share URL only has this param, but it still feels a little too brittle.

ascott replied:
good point. i'll fix this up.

boyers commented:
Thanks @ascott, this is looking good!

My one concern (the reason I didn't build this) is that it might confuse users, since they probably expect using the share URL to automatically start NP—but you still have to click the button to start it manually. Maybe some messaging can help. Curious what you think?

ascott commented:
i thought about the UX as well, and thought that by the time they click on the NP icon, the session gets joined very quickly, and that they would then learn that they just have to hit the NP icon to join the session, but maybe that's not very intuitive.

another option here is to just grab the session id from the url and populate the join session input with it's value. then the user would hit the join session button to join rather than it happening automatically. it would look like this: 
<img width="375" alt="screen shot 2016-02-13 at 3 11 50 pm" src="https://cloud.githubusercontent.com/assets/130878/13030572/90f5335a-d264-11e5-8e11-97b9b666b758.png">

ascott commented:
@boyers i addressed your comments and fixed up the UX. PTAL! :)

![netflix-party-sharing-2](https://cloud.githubusercontent.com/assets/130878/13030809/1994dc0e-d26c-11e5-8dbc-60097b4eeb6e.gif)

boyers commented:
:+1: Thanks @ascott!

boyers commented:
So I thought about it for a bit. I think switching from [NP auto-joins the session from the URL] to [opening NP auto-fills the session ID input from the URL] doesn't actually improve the UX. The former seems better, though both still have the issue that users need to know to click on the icon to join.

So in https://github.com/boyers/netflixparty-chrome/commit/b6ca093b2725754b205cf38039fd1f9d4af144d8, I reverted to the former behavior (auto-join when NP is opened), but added some messaging:

<img width="421" alt="screenshot 2016-02-13 19 53 47" src="https://cloud.githubusercontent.com/assets/796574/13031739/8a387c58-d28b-11e5-8d69-9ced0eb4abab.png">

ascott commented:
:beers:

boyers commented:
<img width="608" alt="screenshot 2016-02-20 20 41 34" src="https://cloud.githubusercontent.com/assets/796574/13200964/82842bb0-d812-11e5-8fe4-522eb7206eab.png">

:sob:

ascott commented:
😿
On Sat, Feb 20, 2016 at 8:45 PM Stephan Boyer <notifications@github.com>
wrote:

> [image: screenshot 2016-02-20 20 41 34]
> <https://cloud.githubusercontent.com/assets/796574/13200964/82842bb0-d812-11e5-8fe4-522eb7206eab.png>
>
> [image: :sob:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/boyers/netflixparty-chrome/pull/4#issuecomment-186740552>
> .
>
